### DISCUSSION TITLE
[DropdownMenu] Possible to specify `boundary` element for PopperJS collision detection?

### DISCUSSION THREAD

**Discussion by ianduvall** [radix-ui/primitives#944] [Ideas]
[PopperJS `boundary` option](https://popper.js.org/docs/v2/utils/detect-overflow/#boundary)

Using this screenshot as an example, I'd like to be able to have the DropdownMenu collision detection calculate based off the gray box "Container" element instead of the viewport. In which case I'd expect the DropdownMenu to open up instead of down.

<img width="843" alt="DropdownMenu component overflowing containing element" src="https://user-images.githubusercontent.com/14840469/139452105-5b0e89ae-26ed-44e6-bed6-95a1bd2c3d20.png">

**Comment by jjenzz** [comment]
We may be able to provide something like this but we do wonder about the UX implications of cases like the following: 

![https://user-images.githubusercontent.com/23072548/136195367-f41043ad-e8fb-4563-93c9-0818f54c4e27.gif](https://user-images.githubusercontent.com/23072548/136195367-f41043ad-e8fb-4563-93c9-0818f54c4e27.gif)

In this case, if the `Page 1` menu boundary was set to its container, it would drop down (off screen) and wouldn't be accessible without scrolling. How would you expect your example to behave in this case if your menu was at the top of your container?

Perhaps we can have some logic that checks if the container boundary is within the viewport and respects it if it can render without collisions. If not, it falls back to viewport boundaries? 🤔  cc: @benoitgrelard

**Reply by benoitgrelard** [reply]
Yeah I have similar questions/concerns. Given that these menus are supposed to behave like os-level menus, on top of everything, using as much as the available area to display them is kinda key, by using a custom container, we'd effectively be reducing that "available" area by a lot, especially if there's overall page scroll involved.

**Reply by ianduvall** [reply]
That's a good point. Thanks for the response.

**Reply by jjenzz** [reply]
> Perhaps we can have some logic that checks if the container boundary is within the viewport and respects it if it can render without collisions. If not, it falls back to viewport boundaries? 🤔 cc: @benoitgrelard

Thoughts on this? It would mean that in the example @ianduvall posted, it would render above until there wasn't enough room and then switch to bottom with viewport as boundary.

**Reply by benoitgrelard** [reply]
Personally I just don't see the value you gain UX-wise if this is how it behaved. If anything I think it would be more confusing.
